src/transformStateWithClones.js

```diff
@@ -7,7 +7,47 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const options = Object.values(actions);
+  const result = [];
+
+  for (const option of options) {
+    const { keysToRemove, type } = option;
+
+    if (type === 'clear') {
+      result.push({});
+    }
+
+    if (type === 'addProperties') {
+      let lastElementOfArray;
+
+      if (result[result.length - 1]) {
+        lastElementOfArray = Object.assign({}, result[result.length - 1],
+          option.extraData);
+      } else {
+        lastElementOfArray = Object.assign({}, state, option.extraData);
+      }
```

Comment:
DRY
You pushed object every time
instead of this create copy `let stateCopy = { ...state };` on the top of function, change it, and push only once on the end of function

```suggestion
     Object.assign(stateCopy, option.extraData);
```